### PR TITLE
Correct conditional for cloud provider

### PR DESCRIPTION
**Pull request by mtnbikenc**
Bug 1578761 | https://bugzilla.redhat.com/show_bug.cgi?id=1578761

### CODE REVIEW

File: roles/openshift_node_group/templates/node-config.yaml.j2 [hunk 1]
```diff
@@ -44,7 +44,7 @@ kubeletArguments:
   - "true"
   cert-dir:
   - /etc/origin/node/certificates
-{% if openshift_node_group_cloud_provider is defined %}
+{% if openshift_node_group_cloud_provider is not none %}
```
Comment (michaelgugino) [inline roles/openshift_node_group/templates/node-config.yaml.j2]: This returns "Not none"
```yaml
---
- hosts: localhost
  connection: local
  vars:
    xxx1: "{{ undefvar1 | default(None) }}"
  tasks:
    - debug:
        msg: "{% if xxx1 is not none %}Not none{% else %}is none{% endif %}"
```

**Comment by michaelgugino** [comment]
@mtnbikenc I recommend we default to empty string ('') and test for that, much more consistent behavior.

**Comment by mtnbikenc** [comment]
Updated.  I wonder if we have bugs hiding in some of the other 147 occurrences of `is not none`.

**Comment by michaelgugino** [comment]
> Updated. I wonder if we have bugs hiding in some of the other 147 occurrences of is not none.

Unfortunately, we most likely do.

**Pull request merged**